src/Game/Connect4/Helpers.elm

```diff
@@ -0,0 +1,128 @@
+module Game.Connect4.Helpers exposing (..)
+
+import Game.Connect4.Types exposing (Player(..), Tile(..))
+import Helpers exposing (chunk, remap)
+
+
+otherPlayer : Player -> Player
+otherPlayer p =
+    case p of
+        X ->
+            O
+
+        O ->
+            X
+
+
+claimedByPlayer : Player -> Tile -> Bool
+claimedByPlayer player tile =
+    case tile of
+        Taken player_ ->
+            player == player_
+
+        _ ->
+            False
+
+
+checkWinner : List Tile -> Maybe Player
+checkWinner tiles =
+    let
+        sets : List (List Tile)
+        sets =
+            List.concatMap (subsets 4) (rows tiles)
+                |> List.append (List.concatMap (subsets 4) (cols tiles))
+                |> List.append (List.concatMap (subsets 4) (diags tiles))
+
+        isWinner : Player -> Bool
+        isWinner player =
+            List.any (List.all (claimedByPlayer player)) sets
+    in
+    case ( isWinner X, isWinner O ) of
+        ( True, _ ) ->
+            Just X
+
+        ( _, True ) ->
+            Just O
+
+        _ ->
+            Nothing
+
+
+subsets : Int -> List a -> List (List a)
+subsets n l =
+    if List.length l < n then
+        []
+
+    else
+        List.take n l :: subsets n (List.drop 1 l)
+
+
+rows : List Tile -> List (List Tile)
+rows tiles =
+    chunk 7 tiles
+
+
+cols : List Tile -> List (List Tile)
+cols tiles =
+    remap tiles
+        [ 0
+        , 7
+        , 14
+        , 21
+        , 28
+        , 35
+        , 1
+        , 8
+        , 15
+        , 22
+        , 29
+        , 36
+        , 2
+        , 9
+        , 16
+        , 23
+        , 30
+        , 37
+        , 3
+        , 10
+        , 17
+        , 24
+        , 31
+        , 38
+        , 4
+        , 11
+        , 18
+        , 25
+        , 32
+        , 39
+        , 5
+        , 12
+        , 19
+        , 26
+        , 33
+        , 40
+        , 6
+        , 13
+        , 20
+        , 27
+        , 34
+        , 41
+        ]
+        |> chunk 6
+
+
+diags : List Tile -> List (List Tile)
```

Comment:
Lol, it took me awhile to come to a decent FP approach on this. I think it's... alright :)